File: problems/121/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices or len(prices) < 2:
            return 0
        
        maxprofit = 0
        minprice = prices[0]
        
        for p in prices:
            maxprofit = max(maxprofit, p - minprice)
            minprice = min(minprice, p)
        
        return maxprofit

File: problems/121/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0, minPrice = INT_MAX;
        
        for (const int& price: prices) {
            if (price > minPrice) {
                ans = max(ans, price - minPrice);
            }
            
            if (price < minPrice) {
                minPrice = price;
            }
        }
        
        return ans;
    }
};